src/format/strftime.rs

```diff
@@ -184,11 +186,6 @@ use alloc::vec::Vec;
 use super::{locales, Locale};
 use super::{Fixed, InternalFixed, InternalInternal, Item, Numeric, Pad};
 
-#[cfg(feature = "unstable-locales")]
```

Comment:
This is the cause of the test failures. Note that, without `unstable-locales`, this is a slice, not a `Vec`. chrono without default features (and in WASM scenarios) has included some formatting functionality in that case, but `Vec` is not available there. So we have to find a way to use a `&[Item<'a>]` in `StrftimeItems`.

Reply:
This could be a problem. I'm assuming that other collection types like `LinkedList` are also unavailable, and I don't see a good way to do this without a resizeable collection.

We could go back to the system where the iterator generates elements as we go and just run through the string once at creation time to check it's all valid. That feels wasteful though.

Reply:
Having spent some time thinking about it, I think we do want to have `StrftimeItems::new` return a `Result` but I can't see any way to use an array as we can't give a size at compile time.
So my current plan is to go back to the system of storing the format string as a `str` and processing it as the iterator is called but iterate through the format string once at creation time to check for problems. Do you have any objections or better ideas?

Reply:
I'll try to look at it again tomorrow.

Reply:
No better ideas, so let's do it that way for now.